### PR TITLE
fastlane env

### PR DESCRIPTION
**Pull request by KrauseFx**
Taken from https://github.com/fastlane/fastlane/pull/6578

screenshot 2016-10-20 11 17 06

### CODE REVIEW

**Comment by KrauseFx** [comment]
This will fail until we have a new fastlane_core release 👍

File: fastlane/lib/fastlane/environment_printer.rb [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+      env_output << "\n\n"
+      env_output
+    end
+
+    # Copy a given string into the clipboard
+    # Make sure to ask the user first, as some people don't
+    # use a clipboard manager, so they might lose something important
+    def self.copy_to_clipboard(string)
+      require 'tmpdir'
+      tmp_file = File.join(Dir.mktmpdir, "temporary") # so that we don't have problems with escaping the string
```
Comment (mfurtak) [inline fastlane/lib/fastlane/environment_printer.rb]: What do you think about:
```ruby
require 'tempfile'
Tempfile.create('environment_printer') do |tmp_file|
  File.write(tmp_file, string)
  `cat '#{tmp_file}' | pbcopy`
end
```

File: fastlane/lib/fastlane/commands_generator.rb [hunk 2]
```diff
@@ -224,6 +227,21 @@ def run
...
       end
 
+      command :env do |c|
+        c.syntax = 'fastlane env'
+        c.description = 'Print your fastlane environment, use this when you submit an issue on GitHub'
+        c.action do |args, options|
+          require "fastlane/environment_printer"
+          env_info = Fastlane::EnvironmentPrinter.get
+          puts env_info
+          if UI.confirm("🙄  Wow, that's a lot of markdown text... should fastlane put it into your clipboard, so you can easily paste it on GitHub?")
```
Comment (mfurtak) [inline fastlane/lib/fastlane/commands_generator.rb]: `pbcopy` will only work for Mac users, so should we avoid showing it to people on other platforms?

**Review by mfurtak — COMMENTED** [review]
Seems great overall - I left two comments.

File: fastlane/lib/fastlane/commands_generator.rb (hunk 2 above)
Comment (KrauseFx) [inline fastlane/lib/fastlane/commands_generator.rb]: Mh, we don't currently support any other platforms, right? But I agree, we probably want to show this only for mac users.

Comment (mfurtak) [inline fastlane/lib/fastlane/commands_generator.rb]: We have definitely tried to make things better for Linux users before 😄

Comment (hjanuschka) [inline fastlane/lib/fastlane/commands_generator.rb]: i think its done with `xclip` on linux  - but has to be checked if installed.

File: fastlane/lib/fastlane/environment_printer.rb (hunk 1 above)
Comment (KrauseFx) [inline fastlane/lib/fastlane/environment_printer.rb]: Thanks, I updated the code. Note that `tmp_file` is not the path, but a `File` object, I had to change the code to use `tmp_file.path` at least on the shelling out part

**Comment by KrauseFx** [comment]
Ready for another review 👍

**Review by mfurtak — APPROVED** [review]

**Pull request merged**